src/ts/localization/i18n.ts

```diff
@@ -0,0 +1,173 @@
+    const translations = this.mergeTranslationsWithDefaultTranslations(config.translations);
+    this.initializeLanguage(config.language, browserLanguageDetection, translations);
+    this.initializeVocabulary(translations);
+  }
+
+  private containsKey(obj: object, key: string) {
+    return obj.hasOwnProperty(key);
+  }
+
+  private mergeTranslationsWithDefaultTranslations(translations: BitmovinPlayerUiTranslations = {}) {
+    const rawTranslations: BitmovinPlayerUiTranslations = { ...defaultTranslations, ...translations };
+    return Object.keys(rawTranslations).reduce((acc, key) => {
+      let translation: CustomVocabulary<Record<string, string>> = rawTranslations[key as string];
+      if (this.containsKey(defaultTranslations, key) && this.containsKey(translations, key)) {
+        translation = { ...defaultTranslations[key], ...translations[key] };
+      }
+      return { ...acc, [key]: translation };
+    }, {});
+  }
+
+  private initializeLanguage(
+    language: string,
+    browserLanguageDetectionEnabled: boolean,
+    translations: BitmovinPlayerUiTranslations,
+  ) {
+    if (browserLanguageDetectionEnabled) {
+      const userLanguage = window.navigator.language;
+
+      if (translations.hasOwnProperty(userLanguage)) {
+        this.language = userLanguage;
+        return;
+      }
+      const shortenedUserLanguage = userLanguage.slice(0, 2);
+      if (translations.hasOwnProperty(shortenedUserLanguage)) {
+        this.language = shortenedUserLanguage;
+        return;
+      }
+    }
+
+    this.language = language;
+  }
+
+  private initializeVocabulary(translations: BitmovinPlayerUiTranslations) {
+    this.vocabulary = ['en', this.language]
+      .reduce((vocab, lang) => ({ ...vocab, ...(translations[lang] || {}) }), {});
+  }
+
+  private replaceVariableWithPlaceholderIfExists(text: string, config: any) {
+    const matches = Array.from(text.match(new RegExp('{[a-zA-Z0-9]+}')));
+    if (matches.length === 0) {
+      return text;
+    }
+
+    return matches
+      .map((m: string) => ({ match: m, key: m.slice(1, -1) }))
+      .reduce((str, { key, match }) => config.hasOwnProperty(key) ? str.replace(match, config[key]) : str, text);
+  }
+
+  public t<V extends CustomVocabulary<Record<string, string>> = CustomVocabulary<Record<string, string>>>(
+    key: keyof V,
+    config?: Record<string, string | number>,
+    ) {
+    return () => {
+      if (key == null) { // because sometimes we call toDomElement() without configuring the component or setting text...
+        return undefined;
+      }
+      let translationString = this.vocabulary[key as string];
+
+      if (translationString == null) {
+        translationString = key as string;
+      }
+
+      if (config != null) {
+        translationString = this.replaceVariableWithPlaceholderIfExists(translationString, config);
+      }
+
+      return translationString;
+    };
+  }
+
+  public getLocalizedText(text: LocalizableText) {
```

Comment:
As this is not really a getter but a function that does some kind of transformation of input to output, it might be better called `localize`!?